mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,19 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+
+        // check if indices is one-based and in ascending order
```

Comment:
indices is -> indices are
You can also use `require` I suppose.
Haven't we lost the check for a index of 0 though now?

You could write this check in one line as something like
`require(indices.sliding(2).forall(p => p(0) < p(1)))`